drv/transceivers-server/src/main.rs

```diff
@@ -320,92 +313,108 @@ impl ServerImpl {
 ////////////////////////////////////////////////////////////////////////////////
 
 impl idl::InOrderTransceiversImpl for ServerImpl {
-    fn get_modules_status(
+    fn get_module_status(
         &mut self,
         _msg: &userlib::RecvMessage,
-    ) -> Result<ModulesStatus, idol_runtime::RequestError<TransceiversError>>
+    ) -> Result<ModuleStatus, idol_runtime::RequestError<TransceiversError>>
     {
-        Ok(self
-            .transceivers
-            .get_modules_status()
-            .map_err(TransceiversError::from)?)
+        let (mod_status, result) = self.transceivers.get_module_status();
+
+        match result.error().count() {
+            0 => Ok(mod_status),
+            _ => Err(RequestError::from(TransceiversError::FpgaError)),
+        }
     }
 
     fn port_enable_power(
         &mut self,
         _msg: &userlib::RecvMessage,
         logical_port_mask: u32,
     ) -> Result<(), idol_runtime::RequestError<TransceiversError>> {
-        self.transceivers
-            .enable_power(LogicalPortMask(logical_port_mask))
-            .map_err(TransceiversError::from)?;
-        Ok(())
+        let mask = LogicalPortMask(logical_port_mask);
+        let result = self.transceivers.enable_power(mask);
+        match (result.error() & mask).count() {
```

Comment:
I don't think the `& mask` is necessary here (or in any of the functions below), since the result should already be masked.

Reply:
Yes and no. All these pin set/clear methods share the failure mode that the only problem we can notice is if FPGA communication is unsuccessful, which will return a `result.error()` mask which corresponds to _all_ the modules on the affected FPGA. So in that case, an operation on `mask=LogicalPortMask(0x1)` would yield a `result.error = LogicalPortMask(0x00ff00ff)`. I was masking that off again because my brain was still in UDP mode where I would actually change my response payload to show an `error` on _only_ the requested ports. In the Idol server this doesn't matter, so I will remove the masking.